parser.ts

```diff
@@ -456,20 +544,30 @@ export function traverseClass(c : TreeCursor, s : string) : Class<SourceLocation
   c.nextSibling(); // Focus on arglist/superclass
   c.nextSibling(); // Focus on body
   c.firstChild();  // Focus colon
-  while(c.nextSibling()) { // Focuses first field
+  var initBody: Array<Stmt<SourceLocation>> = []
+  while (c.nextSibling()) { // Focuses first field
     if (isVarInit(c, s)) {
-      fields.push(traverseVarInit(c, s));
+      let temp = traverseVarInit(c, s)
+      if (JSON.stringify(temp.type) === JSON.stringify({ tag: "class", name: "str" })) {
```

Comment:
Just letting you know 
`JSON.stringify({tag: "class", name: "str"}) === JSON.stringify({name: "str", tag: "class", } )`

returns `false`. So this way of checking your type might be prone to errors. besides, we might have a type that includes `{ tag: "class", name: "str" }` AND some other tags..

Reply:
Now the checks do not depend on stringify but call the equalType function. However, we are not sure if this is the right way to do things since this change be affect many groups.